tools/setup_helpers/generate_code.py

```diff
@@ -51,8 +52,8 @@ def generate_code(ninja_global: Optional[str] = None,
             os.makedirs(d)
     runfiles_dir = os.environ.get("RUNFILES_DIR", None)
     data_dir = os.path.join(runfiles_dir, 'pytorch') if runfiles_dir else ''
-    autograd_dir = os.path.join(data_dir, 'tools', 'autograd')
     tools_jit_templates = os.path.join(data_dir, 'tools', 'jit', 'templates')
+    autograd_dir = os.fspath(pathlib.Path(__file__).parent.parent / "autograd/")
```

Comment:
Wait, is the division operator actually overloaded for Path? haha

Also, is the `/` at the end needed? That might cause issues on windows no?

Reply:
It is overloaded. Cute huh!

Trailing / doesn't seem to do anything with `os.fspath`, at least not in Linux or MacOS.

Example:
```
$ python3 -c 'import os, pathlib; print(os.fspath(pathlib.Path("/dev/null/")))'
/dev/null
```

Reply:
My worry is with windows that uses `\` as separator!
If it does nothing, we should remove it to avoid any issue.

Reply:
removing.